lib/spack/spack/build_environment.py

```diff
@@ -1030,8 +1030,15 @@ def get_cmake_prefix_path(pkg):
                 spack_built.insert(0, dspec)
 
     ordered_build_link_deps = spack_built + externals
-    build_link_prefixes = filter_system_paths(x.prefix for x in ordered_build_link_deps)
-    return build_link_prefixes
+    cmake_prefix_path_entries = []
+    for spec in ordered_build_link_deps:
+        cmake_prefix_path_entries.append(spec.prefix)
+        try:
+            cmake_prefix_path_entries.extend(spec.package.cmake_search_paths)
+        except AttributeError:
+            pass
```

Comment:
If you add a property to `PackageBase`

```
@property
def cmake_prefiix_paths(self):
    return [self.prefix]
```

then this code can be simplified to

```suggestion
    cmake_prefix_path_entries = []
    for spec in ordered_build_link_deps:
        cmake_prefix_path_entries.append(spec.package.cmake_prefix_paths)
```